app/domain/authentication/authn_jwt/fetch_provider_uri_signing_key.rb

```diff
@@ -0,0 +1,75 @@
+module Authentication
+  module AuthnJwt
+    # This class is responsible for fetching JWK Set from provider-uri
+    class FetchProviderUriSigningKey < FetchSigningKeyInterface
```

Comment:
i don't understand. why isn't this a command class? it performs an action.
we typically have 2 kinds of classes:
1. command class to perform actions
2. value objects to hold data

which kind is this class?